mysql/datadog_checks/mysql/mysql.py

```diff
@@ -399,6 +408,57 @@ def _collect_replication_metrics(self, db, results, above_560):
         results.update(self._get_replica_status(db, above_560, nonblocking))
         return REPLICA_VARS
 
+    def _collect_group_replica_metrics(self, db, results):
+        try:
+            with closing(db.cursor()) as cursor:
+                cursor.execute(SQL_GROUP_REPLICATION_MEMBER)
+                replica_results = cursor.fetchone()
+                status = self.CRITICAL
+                additional_tags = []
+                if replica_results is None:
+                    self.log.warning('Unable to get group replica status, setting it as CRITICAL')
+                else:
+                    status = self.OK if replica_results[1] == 'ONLINE' else self.CRITICAL
+                    additional_tags = [
+                        'channel_name:{}'.format(replica_results[0]),
+                        'member_state:{}'.format(replica_results[1]),
+                        'member_role:{}'.format(replica_results[2]),
+                    ]
+                    self.gauge('mysql.replication.group.member_status', 1, tags=additional_tags + self._config.tags)
+
+                self.service_check(
+                    self.GROUP_REPLICATION_SERVICE_CHECK_NAME,
+                    status=status,
+                    tags=self._service_check_tags() + additional_tags,
+                )
+
+                cursor.execute(SQL_GROUP_REPLICATION_METRICS)
+                r = cursor.fetchone()
+
+                if r is None:
+                    self.log.warning('Unable to get group replication metrics')
+                    return {}
+
+                results = {
+                    'Transactions_count': r[1],
+                    'Transactions_check': r[2],
+                    'Conflict_detected': r[3],
+                    'Transactions_row_validating': r[4],
+                    'Transactions_remote_applier_queue': r[5],
+                    'Transactions_remote_applied': r[6],
+                    'Transactions_local_proposed': r[7],
+                    'Transactions_local_rollback': r[8],
+                }
+                # Submit metrics now so it's possible to attach `channel_name` tag
+                self._submit_metrics(
+                    GROUP_REPLICATION_VARS, results, self._config.tags + ['channel_name:{}'.format(r[0])]
+                )
+
+                return GROUP_REPLICATION_VARS
+        except (pymysql.err.InternalError, pymysql.err.OperationalError) as e:
```

Comment:
I think there should be more error handling in this function if we only catch these two types of errors